src/internal/FluenceConnection.ts

```diff
@@ -85,20 +84,25 @@ export class FluenceConnection {
         });
 
         lib2p2Peer.handle([PROTOCOL_NAME], async ({ connection, stream }) => {
-            pipe(stream.source, decode(), async (source: AsyncIterable<string>) => {
-                try {
-                    for await (const msg of source) {
-                        try {
-                            const particle = Particle.fromString(msg);
-                            options.onIncomingParticle(particle);
-                        } catch (e) {
-                            log.error('error on handling a new incoming message: ' + e);
+            pipe(
+                stream.source,
+                // @ts-ignore
+                decode(),
+                async (source: AsyncIterable<string>) => {
+                    try {
+                        for await (const msg of source) {
+                            try {
+                                const particle = Particle.fromString(msg);
+                                options.onIncomingParticle(particle);
+                            } catch (e) {
+                                log.error('error on handling a new incoming message: ' + e);
+                            }
                         }
+                    } catch (e) {
+                        log.debug('connection closed: ' + e);
```

Comment:
Do we need to recreate connection at this point?

Reply:
Probably yes. But this will be addressed in a separate PR